libs/designer-ui/src/lib/tokenpicker/tokenpickersection/tokenpickerhelpers.ts

```diff
@@ -0,0 +1,19 @@
+import type { OutputToken } from '..';
+
+export const getReducedTokenList = (
+  tokens: OutputToken[],
+  options: { hasSearchQuery: boolean; maxRowsShown: number; showAllOptions: boolean }
+): OutputToken[] => {
+  const { hasSearchQuery, maxRowsShown, showAllOptions } = options;
+
+  let filteredTokens = tokens.filter((token) => !token.isAdvanced || showAllOptions || hasSearchQuery);
```

Comment:
The logic change is on lines 9-12. Previously, we checked the index in the filter logic, rather than applying the filter logic _first_ and then taking the first N results.